framework/sources/HFCustomEncoding.m

```diff
@@ -37,6 +38,15 @@ - (BOOL)commonInitWithPath:(NSString *)path {
     if (!identifier) {
         identifier = name;
     }
+    NSNumber *bytesPerCharacter = dict[@"bytesPerCharacter"];
+    if (!bytesPerCharacter) {
```

Comment:
Should add sanity check for `![bytesPerCharacter isKindOfClass:[NSNumber class]]` too.